api-consumption/src/main/java/org/zalando/nakadi/service/subscription/StreamingContext.java

```diff
@@ -277,19 +279,12 @@ public boolean isSubscriptionConsumptionBlocked() {
     }
 
     public boolean isConsumptionBlocked(final ConsumedEvent event) {
```

Comment:
The new code seems incorrect:

1. We should check `isConsumptionBlocked(event)`, even if the event has the matching subscription id tag.
2. If I read it correctly, the new code has the opposite effect — events with subscription id tag are given out to all subscriptions, except the one which should have seen it.

It may be helpful to name the function differently to make it more clear what it should accomplish:
```suggestion
    public boolean shouldSkipEvent(final ConsumedEvent event) {
```

I'm also noticing now that the low-level consumer should check more than just presence of any custom tags to decide if the event should be skipped (namely, it should check the presence of the subscription id tag).